tests/testsuite/artifact_dep.rs

```diff
@@ -436,9 +438,32 @@ fn features_are_not_unified_among_lib_and_bin_dep_of_different_target() {
     p.cargo("build -Z bindeps")
         .masquerade_as_nightly_cargo(&["bindeps"])
         .with_status(101)
-        .with_stderr_contains(
-            "error[E0425]: cannot find function `f2` in crate `d2`\n --> d1/src/main.rs:6:17",
-        )
+        .with_stderr_data(str![[r#"
+[LOCKING] 3 packages to latest compatible versions
+[COMPILING] d2 v0.0.1 ([ROOT]/foo/d2)
+[COMPILING] d1 v0.0.1 ([ROOT]/foo/d1)
+error[E0425]: cannot find function `f2` in crate `d2`
+ --> d1/src/main.rs:6:17
+  |
+6 |             d2::f2();
+  |                 ^^ [HELP] a function with a similar name exists: `f1`
+  |
+ ::: [ROOT]/foo/d2/src/lib.rs:2:42
+  |
+2 |                 #[cfg(feature = "d2f1")] pub fn f1() {}
+  |                                          ----------- similarly named function `f1` defined here
+  |
+[NOTE] found an item that was configured out
+ --> [ROOT]/foo/d2/src/lib.rs:3:49
+  |
+3 |                 #[cfg(feature = "d2f2")] pub fn f2() {}
+  |                                                 ^^
+  = [NOTE] the item is gated behind the `d2f2` feature
+
+For more information about this error, try `rustc --explain E0425`.
+[ERROR] could not compile `d1` (bin "d1") due to 1 previous error
```

Comment:
Should we use `...` to avoid having rustc output in the snapshot?  That seems to cause frustrations in making rustc changed

Reply:
Switched to `...`